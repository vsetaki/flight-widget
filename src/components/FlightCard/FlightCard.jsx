import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { Divider } from '../../../node_modules/@material-ui/core';

const FlightCard = ({
  carrier, arrival, departure, direction,
}) => (
  <Paper>
    <Typography>
      {carrier}
    </Typography>
    <Grid container spacing={16}>
      <Grid item xs={4}>
        {direction.from}
        {departure}
      </Grid>
      <Grid item xs={4}>
        <Divider />
      </Grid>
      <Grid item xs={4}>
        {direction.to}
        {arrival}
      </Grid>
    </Grid>
  </Paper>
);

FlightCard.propTypes = {
  carrier: PropTypes.string.isRequired,
  arrival: PropTypes.string.isRequired,
  departure: PropTypes.string.isRequired,
  direction: PropTypes.shape({
    from: PropTypes.string,
    to: PropTypes.string,
  }).isRequired,
};

export default FlightCard;
