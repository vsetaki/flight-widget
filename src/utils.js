import dayjs from 'dayjs';
import { CARRIER_COLORS } from './constants';

/**
 * Возвращает дату из ISO строки
 * @param {string} dateString
 * @returns {string}
 */
export const getDate = dateString => dayjs(dateString).format('DD.MM.YYYY');

/**
 * Возвращает время из ISO строки
 * @param {string} dateString
 * @returns {string}
 */
export const getTime = dateString => dayjs(dateString).format('HH:mm');

/**
 * Возвращает цвет авиакомпании
 * @param {string} name
 * @returns {string} HEX
 */
export const getCarrierColor = name => CARRIER_COLORS[name];
// в идеале надо регуляркой выбирать цвет
