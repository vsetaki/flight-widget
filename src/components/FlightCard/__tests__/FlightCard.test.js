import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Grid, Card, Typography } from '@material-ui/core';
import FlightIcon from '@material-ui/icons/Flight';
import FlightCard from '../FlightCard';
import flight from '../__fixtures__/FlightCardData';
import { getCarrierColor } from '../../../utils';

Enzyme.configure({ adapter: new Adapter() });


it('renders component', () => {
  const shallowed = shallow(
    <FlightCard
      {...flight}
    />,
  );

  expect(shallowed).toMatchSnapshot();
});

it('contains Card, Grid, Typography, FlightIcon', () => {
  const component = mount(
    <FlightCard
      {...flight}
    />,
  );

  expect(component.containsAllMatchingElements([Card, Grid, Typography, FlightIcon])).toBeTruthy();
});

it('contains flight icon with correct color', () => {
  const component = mount(
    <FlightCard
      {...flight}
    />,
  );
  const color = getCarrierColor(flight.carrier);

  expect(component.find(FlightIcon).prop('style')).toEqual({ fill: color });
});
