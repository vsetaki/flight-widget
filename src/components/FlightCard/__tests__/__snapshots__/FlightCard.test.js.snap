// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<FlightCard
  arrival="2016-06-08T19:52:27.979Z"
  carrier="S7"
  classes={
    Object {
      "card": "FlightCard-card-1",
      "flightIcon": "FlightCard-flightIcon-3",
      "infoBlock": "FlightCard-infoBlock-2",
    }
  }
  departure="2016-06-08T17:51:20.979Z"
  direction={
    Object {
      "from": "Moscow",
      "to": "Berlin",
    }
  }
  id={123}
/>
`;
